src/main/java/it/eng/idsa/dataapp/web/rest/IncomingDataAppResourceOverWs.java

```diff
@@ -55,12 +58,20 @@ public void propertyChange(PropertyChangeEvent evt) {
 			requestedArtifact = reqArtifact.substring(reqArtifact.lastIndexOf('/') + 1);			
 			logger.info("About to get file from " + requestedArtifact);
 			response = readRequestedArtifact(requestMessage, requestedArtifact);
+		} else if (requestMessage instanceof ContractRequestMessage) {
+			response = contractAgreementResponse(requestMessage);
 		} else {
 			response = createDummyResponse(requestHeader);
 		}
 		WebSocketServerManager.getMessageWebSocketResponse().sendResponse(response);
 	}
 
+	private String contractAgreementResponse(Message requestMessage) {
+		MultipartMessage responseMessageMultipart = new MultipartMessageBuilder().withHeaderContent(multiPartMessageService.createContractAgreementMessage((ContractAgreementMessage) requestMessage))
```

Comment:
Please check if this can be formatted, so that each "build" is in new line

Reply:
I will format it right away